sanitycheck: Add method verifying if instance results are filled.

PerMac opened the pull request:
Verifies if results for all test cases in the instance exist.
It can happen that some (or all) test cases are missing in
the instance.results dictionary (due to the instance being filtered,
test code not following sanitycheck's rules etc). It is required for
the full report to be filled correctly.

Signed-off-by: Maciej Perkowski <Maciej.Perkowski@nordicsemi.no>

PerMac committed:
sanitycheck: Add method verifying if instance results are filled.

Verifies if results for all test cases in the instance exist.
It can happen that some (or all) test cases are missing in
the instance.results dictionary (due to the instance being filtered,
test code not following sanitycheck's rules etc). It is required for
the full report to be filled correctly.

Signed-off-by: Maciej Perkowski <Maciej.Perkowski@nordicsemi.no>

PerMac commented:
This patch can lead to some test cases appearing in the full report even though they were not executed nor supposed to be there. But I think it only can happen if the test code is not following the sanitycheck rules. For e.g. if there are #if clauses for different configurations (e.g. tests/kernel/mem_protect/sys_sem .nouser instance does not have test_basic_sem_test test case but sanitycheck does not differentiate this). 
I have found several instances that were not existing in the full report, mostly due to improper test structures (e.g tests/benchmarks/latency_measure) and this patch adds them to the report.
The function from patch can also be modified to give a warning when test cases without coverage in the report are found which can help finding "inaccurate" tests and cleaning them.

LeHack commented:
@PerMac 
> This patch can lead to some test cases appearing in the full report even though they were not executed nor supposed to be there
Well IMO this is bad. We shouldn't be adding false positives into the final report.
I like the idea of filling out blanks, but I'd be more skeptic about the result and mark any test case without a result as blocked/failed or at least skipped *AND* emit a warning every time we encounter such, so it's easy to spot and fix them.

hakehuang commented on scripts/sanity_chk/sanitylib.py:
```diff
@@ -1958,6 +1958,27 @@ def setup_handler(self):
+            for case in self.instance.testcase.cases:
+                if case not in self.instance.results:
+                    self.instance.results[case] = 'SKIP'
+
```
why not just change like this
`

        for case in self.instance.testcase.cases:

            if case not in self.instance.results:

                self.instance.results[case] =self.instance.status.upper
`

hakehuang reviewed (commented):
some code tidy suggestions, function looks ok.

PerMac committed:
sanitycheck: Add method verifying if instance results are filled.

Verifies if results for all test cases in the instance exist.
It can happen that some (or all) test cases are missing in
the instance.results dictionary (due to the instance being filtered,
test code not following sanitycheck's rules etc). It is required for
the full report to be filled correctly.

Signed-off-by: Maciej Perkowski <Maciej.Perkowski@nordicsemi.no>

PerMac commented:
The new commit use only 'SKIP' status for all missing test cases and produce warnings. This should help in cleaning tests which are inconsistent with sanitycheck rules.

maksimmasalski commented:
@PerMac Please rebase
